--- pcc-build/arch/mips/local2.c.orig	2010-10-12 19:13:49.000000000 -0300
+++ pcc-build/arch/mips/local2.c	2010-10-12 19:14:55.000000000 -0300
@@ -103,12 +103,16 @@
 	/* for the moment, just emit this PIC stuff - NetBSD does it */
 	printf("\t.frame %s,%d,%s\n", rnames[FP], ARGINIT/SZCHAR, rnames[RA]);
 	printf("\t.set noreorder\n");
+#if 0
 	printf("\t.cpload $25\t# pseudo-op to load GOT ptr into $25\n");
+#endif
 	printf("\t.set reorder\n");
 
 	printf("\tsubu %s,%s,%d\n", rnames[SP], rnames[SP], ARGINIT/SZCHAR);
+#if 0
 	/* for the moment, just emit PIC stuff - NetBSD does it */
 	printf("\t.cprestore 8\t# pseudo-op to store GOT ptr at 8(sp)\n");
+#endif
 
 	printf("\tsw %s,4(%s)\n", rnames[RA], rnames[SP]);
 	printf("\tsw %s,(%s)\n", rnames[FP], rnames[SP]);
--- pcc-build/arch/mips/code.c.orig	2010-10-12 19:26:55.000000000 -0300
+++ pcc-build/arch/mips/code.c	2010-10-12 19:27:26.000000000 -0300
@@ -435,9 +435,11 @@
 void
 bjobcode()
 {
+#if 0
 	printf("\t.section .mdebug.abi32\n");
 	printf("\t.previous\n");
 	printf("\t.abicalls\n");
+#endif
 }
 
 #ifdef notdef
